models/spring-ai-openai/src/main/java/org/springframework/ai/openai/OpenAiChatModel.java

```diff
@@ -473,7 +473,7 @@ else if (message.getMessageType() == MessageType.TOOL) {
 				updatedRuntimeOptions = ModelOptionsUtils.copyToTarget(((FunctionCallingOptions) prompt.getOptions()),
 						FunctionCallingOptions.class, OpenAiChatOptions.class);
 			}
-			else if (prompt.getOptions() instanceof OpenAiChatOptions) {
+			else {
```

Comment:
This is the same implementation as all the other model integrations. The problem was only in OpenAI.